<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Title:-Group-2-Home-Credit-Exploratory-Data-Analysis---Justin-Hamilton-Contribution" data-toc-modified-id="Title:-Group-2-Home-Credit-Exploratory-Data-Analysis---Justin-Hamilton-Contribution-1">Title: Group 2 Home Credit Exploratory Data Analysis - Justin Hamilton Contribution</a></span></li><li><span><a href="#Question-List" data-toc-modified-id="Question-List-2">Question List</a></span></li><li><span><a href="#Introduction" data-toc-modified-id="Introduction-3">Introduction</a></span></li><li><span><a href="#Description-of-the-Data" data-toc-modified-id="Description-of-the-Data-4">Description of the Data</a></span><ul class="toc-item"><li><span><a href="#Data-Sets-Used-in-Project" data-toc-modified-id="Data-Sets-Used-in-Project-4.1"><strong>Data Sets Used in Project</strong></a></span><ul class="toc-item"><li><span><a href="#application_train.csv" data-toc-modified-id="application_train.csv-4.1.1">application_train.csv</a></span></li><li><span><a href="#application_test.csv" data-toc-modified-id="application_test.csv-4.1.2">application_test.csv</a></span></li><li><span><a href="#bureau.csv" data-toc-modified-id="bureau.csv-4.1.3">bureau.csv</a></span></li><li><span><a href="#bureau_balance.csv" data-toc-modified-id="bureau_balance.csv-4.1.4">bureau_balance.csv</a></span></li><li><span><a href="#credit_card_balance.csv" data-toc-modified-id="credit_card_balance.csv-4.1.5">credit_card_balance.csv</a></span></li><li><span><a href="#HomeCredit_columns_descriptions.csv" data-toc-modified-id="HomeCredit_columns_descriptions.csv-4.1.6">HomeCredit_columns_descriptions.csv</a></span></li><li><span><a href="#installments_payments.csv" data-toc-modified-id="installments_payments.csv-4.1.7">installments_payments.csv</a></span></li><li><span><a href="#POS_CASH_balance.csv" data-toc-modified-id="POS_CASH_balance.csv-4.1.8">POS_CASH_balance.csv</a></span></li><li><span><a href="#previous_application.csv" data-toc-modified-id="previous_application.csv-4.1.9">previous_application.csv</a></span></li><li><span><a href="#sample_submission.csv" data-toc-modified-id="sample_submission.csv-4.1.10">sample_submission.csv</a></span></li></ul></li><li><span><a href="#Steps-Taken" data-toc-modified-id="Steps-Taken-4.2">Steps Taken</a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Installing-Necessary-Packages" data-toc-modified-id="Installing-Necessary-Packages-4.2.0.1">Installing Necessary Packages</a></span></li><li><span><a href="#Importing-Necessary-Packages" data-toc-modified-id="Importing-Necessary-Packages-4.2.0.2">Importing Necessary Packages</a></span></li><li><span><a href="#Making-the-dataframes-for-each-of-our-CSV-files" data-toc-modified-id="Making-the-dataframes-for-each-of-our-CSV-files-4.2.0.3">Making the dataframes for each of our CSV files</a></span></li><li><span><a href="#Printing-the-shape-of-each-of-our-dataframes-for-the-row-and-columns" data-toc-modified-id="Printing-the-shape-of-each-of-our-dataframes-for-the-row-and-columns-4.2.0.4">Printing the shape of each of our dataframes for the row and columns</a></span></li><li><span><a href="#Printing-the-top-5-rows-of-each-dataframe" data-toc-modified-id="Printing-the-top-5-rows-of-each-dataframe-4.2.0.5">Printing the top 5 rows of each dataframe</a></span></li><li><span><a href="#Printing-the-Common-Columns" data-toc-modified-id="Printing-the-Common-Columns-4.2.0.6">Printing the Common Columns</a></span></li><li><span><a href="#Printing-General-Info-for-Each-Dataframe" data-toc-modified-id="Printing-General-Info-for-Each-Dataframe-4.2.0.7">Printing General Info for Each Dataframe</a></span></li><li><span><a href="#Exploring-summary-stats-of-the-numeric-columns" data-toc-modified-id="Exploring-summary-stats-of-the-numeric-columns-4.2.0.8">Exploring summary stats of the numeric columns</a></span></li><li><span><a href="#Exploring-the-categorical-variables-in-the-dataframes" data-toc-modified-id="Exploring-the-categorical-variables-in-the-dataframes-4.2.0.9">Exploring the categorical variables in the dataframes</a></span></li><li><span><a href="#Getting-the-unique-values-for-the-categorical-columns-in-the-dataframe" data-toc-modified-id="Getting-the-unique-values-for-the-categorical-columns-in-the-dataframe-4.2.0.10">Getting the unique values for the categorical columns in the dataframe</a></span></li></ul></li></ul></li></ul></li><li><span><a href="#Justin-Hamilton-Contribution-Summary" data-toc-modified-id="Justin-Hamilton-Contribution-Summary-5">Justin Hamilton Contribution Summary</a></span></li></ul></div>

# Title: Group 2 Home Credit Exploratory Data Analysis - Justin Hamilton Contribution

# Question List

* What is the target variable?
* What is our business and analytic problem?
* What do each of the datasets have in common?
* What information can we draw from in our datasets that will help us understand if a person may be trustworhty or not to borrow?
* Are there any missing variables in our data set that we need to remove?
* What demographic do we want to try and visualize that are in our data?
* How can we use our EDA to lead into our project modeling?

# Introduction

**Project Goal:** Home Credit will be able to identify if a customer is a safe candidate to lend to, then create a personalized customer loan and repayment plan to be accountable for, resulting in an increase in revenue, improved customer experience, and lower default rates.


**Business Problem:** Home Credit desires to know safe borrowers in a customer base that is unfamiliar with banking and give the customer a plan for successful loan repayment. Lending to those who are more likely to default on loans decreases the profits of Home Credit and results in negative customer experiences.

**Analytic Problem:**

The **target variable** is specificially customers that do have a negative history of repayment to lend to, and postive repayment. Represented in the application_train/test.csv sets of binary where 1 = Not trust worthy borrower (Client with payment difficulties), 0 = Trustworthy borrower (client with good repayment history).

* Predict which customers will be good borrowers, using a classification method based on customer financial behavior data.

* Use a regression method to see the relationship that a trust worthy customer has to other attributes about them.



# Description of the Data

## **Data Sets Used in Project**


### application_train.csv
* application_train.csv: 


    * Main dataset. This file contains the information of the target variable as binary for trustworthy to loan to.
   


| Column                          | Description                                                                                          |
|---------------------------------|------------------------------------------------------------------------------------------------------|
| SK_ID_CURR                      | ID of loan in our sample                                                                             |
| TARGET                          | Target variable (1 - client with payment difficulties: he/she had late payment more than X days on at least one of the first Y installments of the loan in our sample, 0 - all other cases)  |
| NAME_CONTRACT_TYPE              | Identification if loan is cash or revolving                                                          |
| CODE_GENDER                     | Gender of the client                                                                                 |
| FLAG_OWN_CAR                    | Flag if the client owns a car                                                                        |
| FLAG_OWN_REALTY                 | Flag if the client owns a house or flat                                                              |
| CNT_CHILDREN                    | Number of children the client has                                                                    |
| AMT_INCOME_TOTAL                | Income of the client                                                                                 |
| AMT_CREDIT                      | Credit amount of the loan                                                                            |
| AMT_ANNUITY                     | Loan annuity                                                                                         |
| AMT_GOODS_PRICE                 | For consumer loans it is the price of the goods for which the loan is given                           |
| NAME_TYPE_SUITE                 | Who was accompanying client when he was applying for the loan                                        |
| NAME_INCOME_TYPE                | Clients income type (businessman, working, maternity leave,...)                                      |
| NAME_EDUCATION_TYPE             | Level of highest education the client achieved                                                       |
| NAME_FAMILY_STATUS              | Family status of the client                                                                          |
| NAME_HOUSING_TYPE               | What is the housing situation of the client (renting, living with parents, ...)                      |
| REGION_POPULATION_RELATIVE      | Normalized population of region where client lives (higher number means the client lives in...       |
| DAYS_BIRTH                      | Client's age in days at the time of application                                                      |
| DAYS_EMPLOYED                   | How many days before the application the person started current employment                           |
| DAYS_REGISTRATION               | How many days before the application did client change his registration                              |
| DAYS_ID_PUBLISH                 | How many days before the application did client change the identity document with which he...        |
| OWN_CAR_AGE                     | Age of client's car                                                                                  |
| FLAG_MOBIL                      | Did client provide mobile phone (1=YES, 0=NO)                                                        |
| FLAG_EMP_PHONE                  | Did client provide work phone (1=YES, 0=NO)                                                          |
| FLAG_WORK_PHONE                 | Did client provide home phone (1=YES, 0=NO)                                                          |
| FLAG_CONT_MOBILE                | Was mobile phone reachable (1=YES, 0=NO)                                                             |
| FLAG_PHONE                      | Did client provide home phone (1=YES, 0=NO)                                                          |
| FLAG_EMAIL                      | Did client provide email (1=YES, 0=NO)                                                               |
| OCCUPATION_TYPE                 | What kind of occupation does the client have                                                         |
| CNT_FAM_MEMBERS                 | How many family members does client have                                                             |
| REGION_RATING_CLIENT            | Our rating of the region where client lives (1,2,3)                                                  |
| REGION_RATING_CLIENT_W_CITY     | Our rating of the region where client lives with taking city into account (1,2,3)                    |
| WEEKDAY_APPR_PROCESS_START      | On which day of the week did the client apply for the loan                                           |
| HOUR_APPR_PROCESS_START         | Approximately at what hour did the client apply for the loan                                         |
| REG_REGION_NOT_LIVE_REGION      | Flag if client's permanent address does not match contact address (1=different, 0=same, at region level        |
| REG_REGION_NOT_WORK_REGION      | Flag if client's permanent address does not match work address (1=different, 0=same, at region level)                                 |
| LIVE_REGION_NOT_WORK_REGION     | Flag if client's contact address does not match work address (1=different, 0=same, at region level)                                    |
| REG_CITY_NOT_LIVE_CITY          | Flag if client's permanent address does not match contact address (1=different, 0=same, at city level)                                 |
| REG_CITY_NOT_WORK_CITY          | Flag if client's permanent address does not match work address (1=different, 0=same, at city level)                                    |
| LIVE_CITY_NOT_WORK_CITY         | Flag if client's contact address does not match work address (1=different, 0=same, at city level)                                       |
| ORGANIZATION_TYPE               | Type of organization where client works                                                                                                 |
| EXT_SOURCE_1                    | Normalized score from external data source                                                                                             |
| EXT_SOURCE_2                    | Normalized score from external data source                                                                                             |
| EXT_SOURCE_3                    | Normalized score from external data source                                                                                             |
| APARTMENTS_AVG                  | Normalized information about building where the client lives, average apartment size, common area, etc.                                |
| BASEMENTAREA_AVG                | Normalized information about building where the client lives, average basement area, common area, etc.                                 |
| YEARS_BEGINEXPLUATATION_AVG     | Normalized information about building where the client lives, average years of building exploitation, etc.                             |
| YEARS_BUILD_AVG                 | Normalized information about building where the client lives, average years since building construction, etc.                          |
| COMMONAREA_AVG                  | Normalized information about building where the client lives, average common area, etc.                                                |
| ELEVATORS_AVG                   | Normalized information about building where the client lives, average number of elevators, etc.                                       |
| ENTRANCES_AVG                   | Normalized information about building where the client lives, average number of entrances, etc.                                       |
| FLOORSMAX_AVG                   | Normalized information about building where the client lives, average maximum number of floors, etc.                                   |
| FLOORSMIN_AVG                   | Normalized information about building where the client lives, average minimum number of floors, etc.                                   |
| LANDAREA_AVG                    | Normalized information about building where the client lives, average land area, etc.                                                  |
| LIVINGAPARTMENTS_AVG            | Normalized information about building where the client lives, average living area of apartments, etc.                                  |
| LIVINGAREA_AVG                  | Normalized information about building where the client lives, average living area, etc.                                                |
| NONLIVINGAPARTMENTS_AVG         | Normalized information about building where the client lives, average non-living area of apartments, etc.                              |
| NONLIVINGAREA_AVG               | Normalized information about building where the client lives, average non-living area, etc.                                            |
| APARTMENTS_MODE                 | Normalized information about building where the client lives, modus (most common) apartment size, common area, etc.                   |
| BASEMENTAREA_MODE               | Normalized information about building where the client lives, modus (most common) basement area, common area, etc.                    |
| YEARS_BEGINEXPLUATATION_MODE    | Normalized information about building |
| YEARS_BUILD_AVG            | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| COMMONAREA_AVG             | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| ELEVATORS_AVG              | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| ENTRANCES_AVG              | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| FLOORSMAX_AVG              | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| FLOORSMIN_AVG              | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| LANDAREA_AVG               | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| LIVINGAPARTMENTS_AVG       | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| LIVINGAREA_AVG             | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| NONLIVINGAPARTMENTS_AVG    | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor   |
| NONLIVINGAREA_AVG         | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| APARTMENTS_MODE           | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| BASEMENTAREA_MODE         | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| YEARS_BEGINEXPLUATATION_MODE | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| YEARS_BUILD_MODE          | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| COMMONAREA_MODE           | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| ELEVATORS_MODE            | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| ENTRANCES_MODE            | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| FLOORSMAX_MODE            | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| FLOORSMIN_MODE            | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| LANDAREA_MODE             | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| LIVINGAPARTMENTS_MODE     | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| LIVINGAREA_MODE           | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| NONLIVINGAPARTMENTS_MODE  | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| NONLIVINGAREA_MODE        | Normalized information about building where the client lives, What is average (_AVG suffix), modus (_MODE suffix), median (_MEDI suffix) apartment size, common area, living area, age of building, number of elevators, number of entrances, state of the building, number of floor |
| FLOORSMIN_MODE            | Normalized information about building where the client lives, such as the mode (most common) value for the number of floors in the building           |
| FLOORSMIN_MEDI            | Normalized information about building where the client lives, such as the median value for the number of floors in the building                   |
| FLOORSMAX_MODE            | Normalized information about building where the client lives, such as the mode (most common) value for the highest floor in the building            |
| FLOORSMAX_MEDI            | Normalized information about building where the client lives, such as the median value for the highest floor in the building                        |
| YEARS_BUILD_MODE          | Normalized information about building where the client lives, such as the mode (most common) value for the age of the building                       |
| YEARS_BUILD_MEDI          | Normalized information about building where the client lives, such as the median value for the age of the building                                   |
| COMMONAREA_MODE           | Normalized information about the common areas in the building where the client lives, such as the mode (most common) value for the common area size |
| COMMONAREA_MEDI           | Normalized information about the common areas in the building where the client lives, such as the median value for the common area size                 |
| ELEVATORS_MODE            | Normalized information about the elevators in the building where the client lives, such as the mode (most common) value for the number of elevators   |
| ELEVATORS_MEDI            | Normalized information about the elevators in the building where the client lives, such as the median value for the number of elevators               |
| ENTRANCES_MODE            | Normalized information about the entrances in the building where the client lives, such as the mode (most common) value for the number of entrances   |
| ENTRANCES_MEDI            | Normalized information about the entrances in the building where the client lives, such as the median value for the number of entrances               |
| LANDAREA_MODE             | Normalized information about the land area of the building where the client lives, such as the mode (most common) value for the land area               |
| LANDAREA_MEDI             | Normalized information about the land area of the building where the client lives, such as the median value for the land area                           |
| LIVINGAPARTMENTS_MODE     | Normalized information about the living apartments in the building where the client lives, such as the mode (most common) value for the apartment size|
| LIVINGAPARTMENTS_MEDI     | Normalized information about the living apartments in the building where the client lives, such as the median value for the apartment size            |
| LIVINGAREA_MODE           | Normalized information about the living area in the building where the client lives, such as the mode (most common) value for the living area size    |
| LIVINGAREA_MEDI           | Normalized information about the living area in the building where the client lives, such as the median value for the living area size                |
| NONLIVINGAPARTMENTS_MODE  | Normalized information about the non-living apartments in the building where the client lives, such as the mode (most common) value for the apartment size|
| NONLIVINGAPARTMENTS_MEDI  | Normalized information about the non-living apartments in the building where the client lives, such as the median value for the apartment size            |
| NONLIVINGAREA_MODE        | Normalized information about the non-living area in the building where the client lives, such as the mode (most common) value for the non-living area size|
| NONLIVINGAREA_MEDI        | Normalized information about the non-living area in the building where the client lives, such as the median value for the non-living area size            |
| FONDKAPREMONT_MODE        | Normalized information about the fund kapremont (capital repairs) mode for the building where the client lives                                      |
| HOUSETYPE_MODE            | Normalized information about the type of house in which the client lives                                                                       |
| TOTALAREA_MODE            | Normalized information about the total area of the building where the client lives, such as the mode (most common) value for the total area            |
| WALLSMATERIAL_MODE        | Normalized information about the wall material used in the building where the client lives                                                     |
| EMERGENCYSTATE_MODE       | Normalized information about the emergency state of the building where the client lives                                                       |
| OBS_30_CNT_SOCIAL_CIRCLE  | Number of observations of clients' social surroundings with observable 30 DPD (days past due) default                                  |
| DEF_30_CNT_SOCIAL_CIRCLE  | Number of clients' social surroundings with 30 DPD (days past due) default                                    |
| OBS_60_CNT_SOCIAL_CIRCLE  | Number of observations of clients' social surroundings with observable 60 DPD (days past due) default                                    |
| DEF_60_CNT_SOCIAL_CIRCLE  | Number of clients' social surroundings with 60 DPD (days past due) default                                    |
| DAYS_LAST_PHONE_CHANGE    | Number of days since the client last changed their phone number                                   |
| FLAG_DOCUMENT_2           | Indicates if there is a second document provided by the client                                   |
| FLAG_DOCUMENT_3           | Indicates if there is a third document provided by the client                                    |
| FLAG_DOCUMENT_4           | Indicates if there is a fourth document provided by the client                                   |
| FLAG_DOCUMENT_5           | Indicates if there is a fifth document provided by the client                                    |
| FLAG_DOCUMENT_6           | Indicates if there is a sixth document provided by the client                                   |
| FLAG_DOCUMENT_7           | Indicates if there is a seventh document provided by the client                                 |
| FLAG_DOCUMENT_8           | Indicates if there is an eighth document provided by the client                                 |
| FLAG_DOCUMENT_9           | Indicates if there is a ninth document provided by the client                                   |
| FLAG_DOCUMENT_10          | Indicates if there is a tenth document provided by the client                                  |
| FLAG_DOCUMENT_11          | Indicates if there is an eleventh document provided by the client                               |
| FLAG_DOCUMENT_12          | Indicates if there is a twelfth document provided by the client                                 |
| FLAG_DOCUMENT_13          | Indicates if there is a thirteenth document provided by the client                              |
| FLAG_DOCUMENT_14          | Indicates if there is a fourteenth document provided by the client                              |
| FLAG_DOCUMENT_15          | Indicates if there is a fifteenth document provided by the client                               |
| FLAG_DOCUMENT_16          | Indicates if there is a sixteenth document provided by the client                               |
| FLAG_DOCUMENT_17          | Indicates if there is a seventeenth document provided by the client                             |
| FLAG_DOCUMENT_18          | Indicates if there is an eighteenth document provided by the client                             |
| FLAG_DOCUMENT_19          | Indicates if there is a nineteenth document provided by the client                              |
| FLAG_DOCUMENT_20          | Indicates if there is a twentieth document provided by the client                               |
| FLAG_DOCUMENT_21          | Indicates if there is a twenty-first document provided by the client                            |
| AMT_REQ_CREDIT_BUREAU_HOUR          | Number of enquiries to Credit Bureau about the client one hour before application |
| AMT_REQ_CREDIT_BUREAU_DAY           | Number of enquiries to Credit Bureau about the client one day before application (excluding one hour before application) |
| AMT_REQ_CREDIT_BUREAU_WEEK          | Number of enquiries to Credit Bureau about the client one week before application (excluding one day before application) |
| AMT_REQ_CREDIT_BUREAU_MON           | Number of enquiries to Credit Bureau about the client one month before application (excluding one week before application) |
| AMT_REQ_CREDIT_BUREAU_QRT           | Number of enquiries to Credit Bureau about the client three months before application (excluding one month before application) |
| AMT_REQ_CREDIT_BUREAU_YEAR          | Number of enquiries to Credit Bureau about the client one year (excluding last three months before application) |




### application_test.csv
* application_test.csv: 

    * Main dataset, but does not include the Target variable information.
    * Same variables as application_train.csv, other no Target variable here.


### bureau.csv
* bureau.csv:
    * Each clients previous credit history provided by the Credit Bureau
    * Each customer has may have multiple rows. Each row is a new credit history that a customer may have had.
    


| Column                    | Description                                                                                                            |
|---------------------------|------------------------------------------------------------------------------------------------------------------------|
| SK_ID_CURR                | ID of loan in our sample - one loan in our sample can have 0, 1, 2 or more related previous credits in credit bureau |
| SK_BUREAU_ID              | Recoded ID of previous Credit Bureau credit related to our loan (unique coding for each loan application)             |
| CREDIT_ACTIVE             | Status of the Credit Bureau (CB) reported credits                                                                     |
| CREDIT_CURRENCY           | Recoded currency of the Credit Bureau credit                                                                           |
| DAYS_CREDIT               | How many days before current application did client apply for Credit Bureau credit                                     |
| CREDIT_DAY_OVERDUE        | Number of days past due on CB credit at the time of application for related loan in our sample                        |
| DAYS_CREDIT_ENDDATE       | Remaining duration of CB credit (in days) at the time of application in Home Credit                                   |
| DAYS_ENDDATE_FACT         | Days since CB credit ended at the time of application in Home Credit (only for closed credit)                         |
| AMT_CREDIT_MAX_OVERDUE    | Maximal amount overdue on the Credit Bureau credit so far (at application date of loan in our sample)                 |
| CNT_CREDIT_PROLONG        | How many times was the Credit Bureau credit prolonged                                                                 |
| AMT_CREDIT_SUM            | Current credit amount for the Credit Bureau credit                                                                    |
| AMT_CREDIT_SUM_DEBT       | Current debt on Credit Bureau credit                                                                                   |
| AMT_CREDIT_SUM_LIMIT      | Current credit limit of credit card reported in Credit Bureau                                                         |
| AMT_CREDIT_SUM_OVERDUE    | Current amount overdue on Credit Bureau credit                                                                         |
| CREDIT_TYPE               | Type of Credit Bureau credit (Car, cash, ...)                                                                          |
| DAYS_CREDIT_UPDATE        | How many days before loan application did last information about the Credit Bureau credit come                        |
| AMT_ANNUITY               | Annuity of the Credit Bureau credit                                                                                    |



### bureau_balance.csv
* bureau_balance.csv:
    * Monthly balances in previous credits from Credit Bureau. 
    * Each row is every month to the Credit Bureau




|        Columns        |                                           Description                                           |
|-----------------------|-----------------------------------------------------------------------------------------------|
| SK_BUREAU_ID          | Recoded ID of Credit Bureau credit (unique coding for each application) - use this to join to CREDIT_BUREAU table |
| MONTHS_BALANCE        | Month of balance relative to application date (-1 means the freshest balance date)              |
| STATUS                | Status of Credit Bureau loan during the month (active, closed, DPD0-30,… [C means closed, X means status unknown, 0 means no DPD, 1 means maximal did during month between 1-30, 2 means DPD 31-60,… 5 means DPD 120+ or sold or written off ]) |



### credit_card_balance.csv
* credit_card_balance.csv:
    * Previous credit cards the customer has had with Home Credit. 
    * One row for each month of history on the previous credit with Home Credit.
    
|              Columns                |                            Description                           |
|-------------------------------------|-----------------------------------------------------------------|
| SK_ID_PREV                          | ID of previous credit in Home Credit related to loan in our sample. (One loan in our sample can have 0, 1, 2 or more previous loans in Home Credit) |
| SK_ID_CURR                          | ID of loan in our sample                                          |
| MONTHS_BALANCE                      | Month of balance relative to application date (-1 means the freshest balance date) |
| AMT_BALANCE                         | Balance during the month of previous credit                       |
| AMT_CREDIT_LIMIT_ACTUAL             | Credit card limit during the month of the previous credit         |
| AMT_DRAWINGS_ATM_CURRENT            | Amount drawing at ATM during the month of the previous credit     |
| AMT_DRAWINGS_CURRENT                | Amount drawing during the month of the previous credit            |
| AMT_DRAWINGS_OTHER_CURRENT          | Amount of other drawings during the month of the previous credit  |
| AMT_DRAWINGS_POS_CURRENT            | Amount drawing or buying goods during the month of the previous credit |
| AMT_INST_MIN_REGULARITY             | Minimal installment for this month of the previous credit         |
| AMT_PAYMENT_CURRENT                 | How much did the client pay during the month on the previous credit |
| AMT_PAYMENT_TOTAL_CURRENT           | How much did the client pay during the month in total on the previous credit |
| AMT_RECEIVABLE_PRINCIPAL            | Amount receivable for principal on the previous credit            |
| AMT_RECIVABLE                       | Amount receivable on the previous credit                          |
| AMT_TOTAL_RECEIVABLE                | Total amount receivable on the previous credit                    |
| CNT_DRAWINGS_ATM_CURRENT            | Number of drawings at ATM during this month on the previous credit |
| CNT_DRAWINGS_CURRENT                | Number of drawings during this month on the previous credit       |
| CNT_DRAWINGS_OTHER_CURRENT          | Number of other drawings during this month on the previous credit |
| CNT_DRAWINGS_POS_CURRENT            | Number of drawings for goods during this month on the previous credit |
| CNT_INSTALMENT_MATURE_CUM           | Number of paid installments on the previous credit               |
| NAME_CONTRACT_STATUS                | Contract status (active signed, ...) on the previous credit       |
| SK_DPD                              | DPD (Days past due) during the month on the previous credit       |
| SK_DPD_DEF                          | DPD (Days past due) during the month with tolerance (debts with low loan amounts are ignored) of the previous credit |


### HomeCredit_columns_descriptions.csv


* HomeCredit_columns_description:
    * Contains descriptions of each of the columns within the datasets. 

### installments_payments.csv
* installments_payments.csv:
    * Repayment history of each of the loans of Home Credit. 
    * One row for every payment that was made
    * A row for each missed payment. 
    
    
|          Columns          |                                Description                                 |
|---------------------------|------------------------------------------------------------------------------|
| SK_ID_PREV                | ID of previous credit in Home credit related to loan in our sample. (One loan in our sample can have 0, 1, 2 or more previous loans in Home Credit) |
| SK_ID_CURR                | ID of loan in our sample                                                    |
| NUM_INSTALMENT_VERSION    | Version of installment calendar (0 is for credit card) of previous credit. Change of installment version from month to month signifies that some parameter of payment calendar has changed |
| NUM_INSTALMENT_NUMBER     | On which installment we observe payment                                      |
| DAYS_INSTALMENT           | When the installment of previous credit was supposed to be paid (relative to application date of current loan) |
| DAYS_ENTRY_PAYMENT        | When the installments of previous credit were actually paid (relative to application date of current loan) |
| AMT_INSTALMENT            | What was the prescribed installment amount of previous credit on this installment |
| AMT_PAYMENT               | What the client actually paid on the previous credit on this installment      |


### POS_CASH_balance.csv
* POS_CASH_balance.csv:
    * Month of the point of sales and cash loans  with Home Credit.
    * One row for each month of history of every previous credit with Home Credit related to each of the loans.
    
    
|         Columns        |                                        Description                                        |
|------------------------|-------------------------------------------------------------------------------------------|
|       SK_ID_PREV       | ID of previous credit in Home Credit related to loan in our sample. (One loan in our sample can have 0, 1, 2, or more previous loans in Home Credit) |
|       SK_ID_CURR       |                         ID of loan in our sample                                          |
|     MONTHS_BALANCE     |           Month of balance relative to application date (-1 means the freshest monthly snapshot, 0 means the information at application)           |
|    CNT_INSTALMENT      |                      Term of previous credit (can change over time)                       |
| CNT_INSTALMENT_FUTURE  |                 Installments left to pay on the previous credit                           |
| NAME_CONTRACT_STATUS   |                           Contract status during the month                                |
|         SK_DPD         |                 DPD (days past due) during the month of previous credit                    |
|       SK_DPD_DEF       |    DPD during the month with tolerance (debts with low loan amounts are ignored)          |


### previous_application.csv
* previous_application.csv:
    * All previous loan applications of customers with Home Credit
    
    
|             Columns             |                                      Description                                      |
|---------------------------------|----------------------------------------------------------------------------------------|
| SK_ID_PREV                      | ID of previous credit in Home Credit related to loan in our sample                      |
| SK_ID_CURR                      | ID of loan in our sample                                                               |
| NAME_CONTRACT_TYPE              | Contract product type (Cash loan, consumer loan [POS], ...) of the previous application |
| AMT_ANNUITY                     | Annuity of previous application                                                        |
| AMT_APPLICATION                 | For how much credit did the client ask on the previous application                       |
| AMT_CREDIT                      | Final credit amount on the previous application                                        |
| AMT_DOWN_PAYMENT                | Down payment on the previous application                                               |
| AMT_GOODS_PRICE                 | Goods price of good that the client asked for (if applicable) on the previous application |
| WEEKDAY_APPR_PROCESS_START      | On which day of the week did the client apply for the previous application              |
| HOUR_APPR_PROCESS_START         | Approximately at what hour of the day did the client apply for the previous application |
| FLAG_LAST_APPL_PER_CONTRACT     | Flag if it was the last application for the previous contract                           |
| NFLAG_LAST_APPL_IN_DAY          | Flag if the application was the last application per day of the client                  |
| NFLAG_MICRO_CASH                | Flag for Micro finance loan                                                            |
| RATE_DOWN_PAYMENT               | Down payment rate normalized on the previous credit                                    |
| RATE_INTEREST_PRIMARY           | Interest rate normalized on the previous credit                                        |
| RATE_INTEREST_PRIVILEGED        | Interest rate normalized on the previous credit                                        |
| NAME_CASH_LOAN_PURPOSE          | Purpose of the cash loan                                                               |
| NAME_CONTRACT_STATUS            | Contract status (approved, cancelled, ...) of the previous application                 |
| DAYS_DECISION                   | Relative to the current application, when was the decision about the previous application made |
| NAME_PAYMENT_TYPE               | Payment method that the client chose to pay for the previous application               |
| CODE_REJECT_REASON              | Reason why the previous application was rejected                                       |
| NAME_TYPE_SUITE                 | Who accompanied the client when applying for the previous application                  |
| NAME_CLIENT_TYPE                | Whether the client was an old or new client when applying for the previous application |
| NAME_GOODS_CATEGORY             | What kind of goods did the client apply for in the previous application                 |
| NAME_PORTFOLIO                  | Whether the previous application was for CASH, POS, CAR, etc.                          |
| NAME_PRODUCT_TYPE               | Whether the previous application was x-sell or walk-in                                 |
| CHANNEL_TYPE                    | Through which channel we acquired the client on the previous application               |
| SELLERPLACE_AREA                | Selling area of the seller place of the previous application                           |
| NAME_SELLER_INDUSTRY            | The industry of the seller                                                            |
| CNT_PAYMENT                     | Term of the previous credit at the application of the previous application             |
| NAME_YIELD_GROUP                | Grouped interest rate into small, medium, and high of the previous application         |
| PRODUCT_COMBINATION             | Detailed product combination of the previous application                               |
| DAYS_FIRST_DRAWING              | Relative to the application date of the current application, when was the first disbursement of the previous application |
| DAYS_FIRST_DUE                  | Relative to the application date of the current application, when was the first due supposed to be of the previous application |
| DAYS_LAST_DUE_1ST_VERSION       | Relative to the application date of the current application, when was the first due of the previous application |
| DAYS_LAST_DUE                   | Relative to the application date of the current application, when was the last due date of the previous application |
| DAYS_TERMINATION                | Relative to the application date of the current application, when was the expected termination of the previous application |
| NFLAG_INSURED_ON_APPROVAL       | Whether the client requested insurance during the previous application                  |


### sample_submission.csv
* sample_submission.csv:
    * A sample of classifying each of the customers using a classification based on the Target variable of being Trustworthy to loan to or not. 

## Steps Taken

#### Installing Necessary Packages

In [1]:
# !pip install pandas
# !pip install numpy

#### Importing Necessary Packages

In [2]:
import pandas as pd
import numpy as np

#### Making the dataframes for each of our CSV files

In [3]:
# Making Dataframes of each of the CSV Files

applicationtestDF = pd.read_csv("application_test.csv")
applicationtrainDF = pd.read_csv("application_train.csv")
bureauDF = pd.read_csv("bureau.csv")
bureaubalanceDF = pd.read_csv("bureau_balance.csv")
creditcardbalanceDF = pd.read_csv("credit_card_balance.csv")
installmentpaymentsDF = pd.read_csv("installments_payments.csv")
poscashDF = pd.read_csv("POS_CASH_balance.csv")
previousDF = pd.read_csv("previous_application.csv")

#### Printing the shape of each of our dataframes for the row and columns

In [4]:
# Printing the shape of each of the dataframes
dataframes = [applicationtestDF, applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, installmentpaymentsDF, poscashDF, previousDF]

for df in dataframes:b
    print(df.shape)


(48744, 121)
(307511, 122)
(1716428, 17)
(27299925, 3)
(3840312, 23)
(13605401, 8)
(10001358, 8)
(1670214, 37)


We see here that the applicationtrainDF contains the most variables that we can use for data analysis. We also see that the creditcardbalanceDF and previousDF contain the most variables, which could be interesting to explore in our analysis

#### Printing the top 5 rows of each dataframe

In [5]:
# Printing the head of each of the dataframes
dataframes = [applicationtestDF, applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, installmentpaymentsDF, poscashDF, previousDF]

for df in dataframes:
    print(df.head(n=5))

    

   SK_ID_CURR NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  \
0      100001         Cash loans           F            N               Y   
1      100005         Cash loans           M            N               Y   
2      100013         Cash loans           M            Y               Y   
3      100028         Cash loans           F            N               Y   
4      100038         Cash loans           M            Y               N   

   CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE  \
0             0          135000.0    568800.0      20560.5         450000.0   
1             0           99000.0    222768.0      17370.0         180000.0   
2             0          202500.0    663264.0      69777.0         630000.0   
3             2          315000.0   1575000.0      49018.5        1575000.0   
4             1          180000.0    625500.0      32067.0         625500.0   

   ... FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG

With a quick scroll through, we can see that each of the dataframes carry same or similar variables. In the next step we see exactly of the data sets which variables are common across each of the datasets. 

#### Printing the Common Columns

In [6]:
# Printing out the common columns amongst the dataframes

datasets = [applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, installmentpaymentsDF, poscashDF, previousDF]


# Get the column names from each DataFrame
columns = [set(df.columns.str.strip().str.lower().tolist()) for df in dataframes]

# Initialize a list to store the common columns for each comparison
common_columns = []

# Iterate over the dataframes
for i in range(len(dataframes)):
    for j in range(i + 1, len(dataframes)):
        df1 = dataframes[i]
        df2 = dataframes[j]
        
        # Get the column names for the current pair of dataframes
        columns_df1 = columns[i]
        columns_df2 = columns[j]
        
        # Find the common columns between the two dataframes
        common = columns_df1.intersection(columns_df2)
        
        # Append the common columns for the current pair
        common_columns.append((common))

# Print the common column names for each pair of dataframes
for common in common_columns:
    if len(common) == 0:
        print("No common columns found.")
    else:
        print(", ".join(common))




flag_document_18, flag_email, cnt_fam_members, flag_document_13, name_income_type, name_education_type, def_30_cnt_social_circle, sk_id_curr, flag_document_12, years_build_avg, entrances_avg, wallsmaterial_mode, years_build_mode, reg_region_not_live_region, region_population_relative, fondkapremont_mode, years_beginexpluatation_medi, housetype_mode, obs_30_cnt_social_circle, totalarea_mode, live_city_not_work_city, ext_source_3, days_last_phone_change, flag_emp_phone, flag_document_21, amt_req_credit_bureau_year, reg_city_not_work_city, flag_document_11, flag_cont_mobile, region_rating_client, amt_goods_price, region_rating_client_w_city, flag_document_17, flag_document_20, nonlivingapartments_mode, years_build_medi, flag_mobil, floorsmin_medi, days_employed, nonlivingapartments_avg, flag_document_7, name_housing_type, flag_document_3, name_contract_type, elevators_medi, own_car_age, elevators_avg, nonlivingarea_avg, years_beginexpluatation_mode, entrances_medi, nonlivingapartments_med

From printing the common colunmns across the dataframes, we see that many of them have the ID, which is a useful variable that in the future we could use to merge across the dataframes. 

We also see that other variables that share common columns amongst each set is the name_contract_status, name_contract_type, each of these may be variables that we can use to manipulate the datasets. It also shows that these are variables that carry a lot of data with them. 

#### Printing General Info for Each Dataframe

In [7]:
# Printing info about each of the dataframes
dataframes = [applicationtestDF, applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, installmentpaymentsDF, poscashDF, previousDF]

for df in dataframes:
    display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48744 entries, 0 to 48743
Columns: 121 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(40), object(16)
memory usage: 45.0+ MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), object(16)
memory usage: 286.2+ MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1716428 entries, 0 to 1716427
Data columns (total 17 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   SK_ID_CURR              int64  
 1   SK_ID_BUREAU            int64  
 2   CREDIT_ACTIVE           object 
 3   CREDIT_CURRENCY         object 
 4   DAYS_CREDIT             int64  
 5   CREDIT_DAY_OVERDUE      int64  
 6   DAYS_CREDIT_ENDDATE     float64
 7   DAYS_ENDDATE_FACT       float64
 8   AMT_CREDIT_MAX_OVERDUE  float64
 9   CNT_CREDIT_PROLONG      int64  
 10  AMT_CREDIT_SUM          float64
 11  AMT_CREDIT_SUM_DEBT     float64
 12  AMT_CREDIT_SUM_LIMIT    float64
 13  AMT_CREDIT_SUM_OVERDUE  float64
 14  CREDIT_TYPE             object 
 15  DAYS_CREDIT_UPDATE      int64  
 16  AMT_ANNUITY             float64
dtypes: float64(8), int64(6), object(3)
memory usage: 222.6+ MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27299925 entries, 0 to 27299924
Data columns (total 3 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   SK_ID_BUREAU    int64 
 1   MONTHS_BALANCE  int64 
 2   STATUS          object
dtypes: int64(2), object(1)
memory usage: 624.8+ MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3840312 entries, 0 to 3840311
Data columns (total 23 columns):
 #   Column                      Dtype  
---  ------                      -----  
 0   SK_ID_PREV                  int64  
 1   SK_ID_CURR                  int64  
 2   MONTHS_BALANCE              int64  
 3   AMT_BALANCE                 float64
 4   AMT_CREDIT_LIMIT_ACTUAL     int64  
 5   AMT_DRAWINGS_ATM_CURRENT    float64
 6   AMT_DRAWINGS_CURRENT        float64
 7   AMT_DRAWINGS_OTHER_CURRENT  float64
 8   AMT_DRAWINGS_POS_CURRENT    float64
 9   AMT_INST_MIN_REGULARITY     float64
 10  AMT_PAYMENT_CURRENT         float64
 11  AMT_PAYMENT_TOTAL_CURRENT   float64
 12  AMT_RECEIVABLE_PRINCIPAL    float64
 13  AMT_RECIVABLE               float64
 14  AMT_TOTAL_RECEIVABLE        float64
 15  CNT_DRAWINGS_ATM_CURRENT    float64
 16  CNT_DRAWINGS_CURRENT        int64  
 17  CNT_DRAWINGS_OTHER_CURRENT  float64
 18  CNT_DRAWINGS_POS_CURRENT    float64
 19  CNT_INSTALMENT_MATURE

None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13605401 entries, 0 to 13605400
Data columns (total 8 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   SK_ID_PREV              int64  
 1   SK_ID_CURR              int64  
 2   NUM_INSTALMENT_VERSION  float64
 3   NUM_INSTALMENT_NUMBER   int64  
 4   DAYS_INSTALMENT         float64
 5   DAYS_ENTRY_PAYMENT      float64
 6   AMT_INSTALMENT          float64
 7   AMT_PAYMENT             float64
dtypes: float64(5), int64(3)
memory usage: 830.4 MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10001358 entries, 0 to 10001357
Data columns (total 8 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   SK_ID_PREV             int64  
 1   SK_ID_CURR             int64  
 2   MONTHS_BALANCE         int64  
 3   CNT_INSTALMENT         float64
 4   CNT_INSTALMENT_FUTURE  float64
 5   NAME_CONTRACT_STATUS   object 
 6   SK_DPD                 int64  
 7   SK_DPD_DEF             int64  
dtypes: float64(2), int64(5), object(1)
memory usage: 610.4+ MB


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1670214 entries, 0 to 1670213
Data columns (total 37 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   SK_ID_PREV                   1670214 non-null  int64  
 1   SK_ID_CURR                   1670214 non-null  int64  
 2   NAME_CONTRACT_TYPE           1670214 non-null  object 
 3   AMT_ANNUITY                  1297979 non-null  float64
 4   AMT_APPLICATION              1670214 non-null  float64
 5   AMT_CREDIT                   1670213 non-null  float64
 6   AMT_DOWN_PAYMENT             774370 non-null   float64
 7   AMT_GOODS_PRICE              1284699 non-null  float64
 8   WEEKDAY_APPR_PROCESS_START   1670214 non-null  object 
 9   HOUR_APPR_PROCESS_START      1670214 non-null  int64  
 10  FLAG_LAST_APPL_PER_CONTRACT  1670214 non-null  object 
 11  NFLAG_LAST_APPL_IN_DAY       1670214 non-null  int64  
 12  RATE_DOWN_PAYMENT            774370 non-nu

None

For each dataset, seeing the data type of each variable is important in letting us understand more of what each of the columns are. Identifying each of these data types helps us referrence on how we best interpret the results that we get.

#### Exploring summary stats of the numeric columns

In [8]:
# Get summary statistics of the numerical columns
dataframes = [applicationtestDF, applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, installmentpaymentsDF, poscashDF, previousDF]

for df in dataframes:
    print(df.describe())

          SK_ID_CURR  CNT_CHILDREN  AMT_INCOME_TOTAL    AMT_CREDIT  \
count   48744.000000  48744.000000      4.874400e+04  4.874400e+04   
mean   277796.676350      0.397054      1.784318e+05  5.167404e+05   
std    103169.547296      0.709047      1.015226e+05  3.653970e+05   
min    100001.000000      0.000000      2.694150e+04  4.500000e+04   
25%    188557.750000      0.000000      1.125000e+05  2.606400e+05   
50%    277549.000000      0.000000      1.575000e+05  4.500000e+05   
75%    367555.500000      1.000000      2.250000e+05  6.750000e+05   
max    456250.000000     20.000000      4.410000e+06  2.245500e+06   

         AMT_ANNUITY  AMT_GOODS_PRICE  REGION_POPULATION_RELATIVE  \
count   48720.000000     4.874400e+04                48744.000000   
mean    29426.240209     4.626188e+05                    0.021226   
std     16016.368315     3.367102e+05                    0.014428   
min      2295.000000     4.500000e+04                    0.000253   
25%     17973.000000    

         SK_ID_CURR  SK_ID_BUREAU   DAYS_CREDIT  CREDIT_DAY_OVERDUE  \
count  1.716428e+06  1.716428e+06  1.716428e+06        1.716428e+06   
mean   2.782149e+05  5.924434e+06 -1.142108e+03        8.181666e-01   
std    1.029386e+05  5.322657e+05  7.951649e+02        3.654443e+01   
min    1.000010e+05  5.000000e+06 -2.922000e+03        0.000000e+00   
25%    1.888668e+05  5.463954e+06 -1.666000e+03        0.000000e+00   
50%    2.780550e+05  5.926304e+06 -9.870000e+02        0.000000e+00   
75%    3.674260e+05  6.385681e+06 -4.740000e+02        0.000000e+00   
max    4.562550e+05  6.843457e+06  0.000000e+00        2.792000e+03   

       DAYS_CREDIT_ENDDATE  DAYS_ENDDATE_FACT  AMT_CREDIT_MAX_OVERDUE  \
count         1.610875e+06       1.082775e+06            5.919400e+05   
mean          5.105174e+02      -1.017437e+03            3.825418e+03   
std           4.994220e+03       7.140106e+02            2.060316e+05   
min          -4.206000e+04      -4.202300e+04            0.000000e+0

         SK_ID_PREV    SK_ID_CURR  MONTHS_BALANCE  CNT_INSTALMENT  \
count  1.000136e+07  1.000136e+07    1.000136e+07    9.975287e+06   
mean   1.903217e+06  2.784039e+05   -3.501259e+01    1.708965e+01   
std    5.358465e+05  1.027637e+05    2.606657e+01    1.199506e+01   
min    1.000001e+06  1.000010e+05   -9.600000e+01    1.000000e+00   
25%    1.434405e+06  1.895500e+05   -5.400000e+01    1.000000e+01   
50%    1.896565e+06  2.786540e+05   -2.800000e+01    1.200000e+01   
75%    2.368963e+06  3.674290e+05   -1.300000e+01    2.400000e+01   
max    2.843499e+06  4.562550e+05   -1.000000e+00    9.200000e+01   

       CNT_INSTALMENT_FUTURE        SK_DPD    SK_DPD_DEF  
count           9.975271e+06  1.000136e+07  1.000136e+07  
mean            1.048384e+01  1.160693e+01  6.544684e-01  
std             1.110906e+01  1.327140e+02  3.276249e+01  
min             0.000000e+00  0.000000e+00  0.000000e+00  
25%             3.000000e+00  0.000000e+00  0.000000e+00  
50%             7.000000

Based on this output, some are key findings are that

* The average income of applicants is around 178,432 dollars. However, the high standard deviation indicates that there is significant variability in income levels, with values ranging from 26,941.50 dollars to 4,410,000 dollars.

* The average credit amount applied for by clients is approximately 516,740 dollars. The standard deviation suggests that credit amounts vary widely, with values ranging from 45,000 to 2,245,500.

* The average annuity amount is approximately 29,426 dollars. The standard deviation suggests some variability in annuity amounts, ranging from 2,295 to 180,576 dollars.

#### Exploring the categorical variables in the dataframes

In [9]:
# Getting information about the categorical columns
# installmentpaymentsDF does not include any categorical data so we do not include it.

dataframes = [applicationtestDF, applicationtrainDF, bureauDF, bureaubalanceDF, creditcardbalanceDF, poscashDF, previousDF]

for df in dataframes:
    print(df.describe(include='object'))

       NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  \
count               48744       48744        48744           48744   
unique                  2           2            2               2   
top            Cash loans           F            N               Y   
freq                48305       32678        32311           33658   

       NAME_TYPE_SUITE NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
count            47833            48744                          48744   
unique               7                7                              5   
top      Unaccompanied          Working  Secondary / secondary special   
freq             39727            24533                          33988   

       NAME_FAMILY_STATUS  NAME_HOUSING_TYPE OCCUPATION_TYPE  \
count               48744              48744           33139   
unique                  5                  6              18   
top               Married  House / apartment        Laborers   
freq                32

From this output, some interesting interpretations of the categorical variables could be that 

* "Cash loans" are more frequent, with 48,305 occurrences out of 48,744, indicating that most contracts fall as cash loans. Which may become an important feature on if the client will default.

* The most frequent gender is female (F), with 32,678 occurrences out of 48,744. Helping us later understand if gender is an important feature in if a client will default.

* There are five unique values representing different levels of education, with "Secondary / secondary special" being the most common one. The most frequent education type is "Secondary / secondary special," occurring 33,988 times out of 48,744. This may lead us to find that potentially those that are pursuing a secondary education such as college, are more likely to default than those who are not.

#### Getting the unique values for the categorical columns in the dataframe

In [10]:
# Get the unique values in each categorical column
def get_unique_values(data):
    unique_values = {}
    for column in data.select_dtypes(include='object'):
        unique_values[column] = data[column].unique()

    for column, values in unique_values.items():
        print("\n")
        print(f"{column}: {values}")

        
# Example usage with multiple DataFrames        
get_unique_values(applicationtestDF)
get_unique_values(applicationtrainDF)
get_unique_values(bureauDF)
get_unique_values(bureaubalanceDF)
get_unique_values(creditcardbalanceDF)
get_unique_values(installmentpaymentsDF)
get_unique_values(poscashDF)
get_unique_values(previousDF)




NAME_CONTRACT_TYPE: ['Cash loans' 'Revolving loans']


CODE_GENDER: ['F' 'M']


FLAG_OWN_CAR: ['N' 'Y']


FLAG_OWN_REALTY: ['Y' 'N']


NAME_TYPE_SUITE: ['Unaccompanied' nan 'Family' 'Spouse, partner' 'Group of people'
 'Other_B' 'Children' 'Other_A']


NAME_INCOME_TYPE: ['Working' 'State servant' 'Pensioner' 'Commercial associate'
 'Businessman' 'Student' 'Unemployed']


NAME_EDUCATION_TYPE: ['Higher education' 'Secondary / secondary special' 'Incomplete higher'
 'Lower secondary' 'Academic degree']


NAME_FAMILY_STATUS: ['Married' 'Single / not married' 'Civil marriage' 'Widow' 'Separated']


NAME_HOUSING_TYPE: ['House / apartment' 'With parents' 'Rented apartment'
 'Municipal apartment' 'Office apartment' 'Co-op apartment']


OCCUPATION_TYPE: [nan 'Low-skill Laborers' 'Drivers' 'Sales staff' 'High skill tech staff'
 'Core staff' 'Laborers' 'Managers' 'Accountants' 'Medicine staff'
 'Security staff' 'Private service staff' 'Secretaries' 'Cleaning staff'
 'Cooking staff' 'HR staff' '

This output helps us to understand each of the variables better. We are able to see the full picture for each of the variables to know exactly which categories are possible for a profile to be under as for each categorical varaible. 

Knowing all of the unique values will help us paint a bigger picture of understanding and intepreting each of the variables and their connection with a person defaulting on a loan.

We see that many of these do contain the value of Naan, so they will need to be cleaned for a better analysis.

# Justin Hamilton Contribution Summary

Justin worked on the exploration and description of the dataset. Justin first started by creating tables that had the data descriptions from Kaggle. Justin then created each of the data frames by reading in the CSVs. Then using the data frames created, he viewed their shape, then viewing each of  the top 5 rows of the data frames. He then used a function to view all of the common columns that appeared multiple times across the data frames. He then examined the categorical vs numeric data that was found within each data frame by looking at the summary stats of the numeric data and then looking at the proportions and unique values of the categorical data to see a vector of all possible classifications for categorical variables. 
